lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemOutputFile.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.exception.AlluxioException;
+import alluxio.grpc.CreateFilePOptions;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoOutputFile;
+import io.trino.memory.context.AggregatedMemoryContext;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.FileAlreadyExistsException;
+
+import static io.trino.filesystem.alluxio.AlluxioUtils.convertToAlluxioURI;
+import static java.util.Objects.requireNonNull;
+
+public class AlluxioFileSystemOutputFile
+        implements TrinoOutputFile
+{
+    private final Location rootLocation;
+
+    private final Location location;
+
+    private final FileSystem fileSystem;
+
+    private final String mountRoot;
+
+    public AlluxioFileSystemOutputFile(Location rootLocation, Location location, FileSystem fileSystem, String mountRoot)
+    {
+        this.rootLocation = rootLocation;
+        this.location = requireNonNull(location, "location is null");
+        this.fileSystem = requireNonNull(fileSystem, "fileSystem is null");
+        this.mountRoot = requireNonNull(mountRoot, "mountRoot is null");
+    }
+
+    @Override
+    public void createExclusive(byte[] data)
+            throws IOException
+    {
+        throwIfAlreadyExists();
+        try (FileOutStream outStream =
+                     fileSystem.createFile(convertToAlluxioURI(location, mountRoot), CreateFilePOptions.newBuilder().setRecursive(true).setOverwrite(true).build())) {
```

Comment:
createFile can go on previous line and each argument on newline

Reply:
resolved